nixos/modules/services/home-automation/zwave-js-ui.nix

```
@@ -0,0 +1,105 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+let
+  inherit (lib)
+    mkIf
+    mkEnableOption
+    mkOption
+    mkPackageOption
+    types
+    ;
+  cfg = config.services.zwave-js-ui;
+in
+{
+  options.services.zwave-js-ui = {
+    enable = mkEnableOption "zwave-js-ui";
+
+    package = mkPackageOption pkgs "zwave-js-ui" { };
+
+    serialPort = mkOption {
+      type = types.path;
+      description = ''
+        Serial port for the Z-Wave controller.
+
+        Only used to grant permissions to the device; must be additionally configured in the application
+      '';
+      example = "/dev/serial/by-id/usb-example";
+    };
+
+    settings = mkOption {
+      type =
+        with types;
+        attrsOf (
+          nullOr (oneOf [
+            str
+            path
+            package
+          ])
+        );
+      description = ''
+        Extra environment entries for the zwave-js-ui process.
+
+        Check https://zwave-js.github.io/zwave-js-ui/#/guide/env-vars for more information
+      '';
+      example = {
+        HOST = "::";
+        PORT = "8091";
+      };
+    };
+  };
+  config = mkIf cfg.enable {
+    systemd.services.zwave-js-ui = {
+      environment = {
+        STORE_DIR = "%S/zwave-js-ui";
+        ZWAVEJS_EXTERNAL_CONFIG = "%S/zwave-js-ui/.config-db";
+      } // cfg.settings;
+      wantedBy = [ "multi-user.target" ];
+      serviceConfig = {
+        ExecStart = "${cfg.package}/bin/zwave-js-ui";
```

```suggestion
        ExecStart = lib.getExe cfg.package;
```

Reuse `meta.mainProgram`.